exporter/googlecloudexporter/factory.go

```diff
@@ -38,6 +40,10 @@ func NewFactory() component.ExporterFactory {
 	// register view for self-observability
 	once.Do(func() {
 		view.Register(viewPointCount)
+		// Register views to collect data.
+		if err := view.Register(ocgrpc.DefaultClientViews...); err != nil {
+			log.Fatal(err)
```

Comment:
Two things:

- Please use zap.Logger instead of the std log implementation.
- Registering the view's here is problematic since it would break on application start up if there was an issue.

Can I ask if this call to Register be done elsewhere where the error can be handled correctly?

Reply:
It can't be done elsewhere... 
Initially, i put the view.Register(ocgrpcx...) inside googecloud.go.By doing that, the agent didn't generate any grpc metrics. After going thru how other exporters deal with agent exporter metrics, i found out they always put the view.Register inside the component.ExporterFactory, only by doing this, i tested it started to generate grpc metrics. Sample on how other exporter handle such cases https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/newrelicexporter/factory.go#L33

I can modify it to 
```
	once.Do(func() {
		view.Register(viewPointCount)
                view.Register(ocgrpc.DefaultClientViews...)
```

Reply:
That sounds reasonable to me, the only reason i am hesitant is that with the correct approach is it can cause panic on start up of no fault to the user and would require either downgrading the collector or trying to rollout a fix for it.

Reply:
Done removing panicking :)